Code/GraphMol/Wrap/rough_test.py

```diff
@@ -5051,6 +5051,34 @@ def testEnhancedStereoPreservesMol(self):
     self.assertEqual(stereo_atoms[1].GetIdx(), 4)
     self.assertEqual(stereo_atoms[1].GetOwningMol().GetNumAtoms(),8)
 
+  def testSetEnhancedStereoGroup(self):
+    m = Chem.MolFromSmiles('F[C@@H](Br)[C@H](F)Cl |o1:1|')
+    m2 = Chem.RWMol(m)
+
+    groups = m2.GetStereoGroups()
+    self.assertEqual(len(groups), 1)
+    # Can set the StereoGroups using an empty list
+    m2.SetStereoGroups([])
+    self.assertEqual(len(m2.GetStereoGroups()), 0)
+
+    # Can create a StereoGroup using a Python list
+    group1 = Chem.rdchem.StereoGroup(Chem.rdchem.StereoGroupType.STEREO_OR, [m.GetAtomWithIdx(1)])
+    # Can set the StereoGroups using a Python list
+    m2.SetStereoGroups([group1])
```

Comment:
This highlights one of the things I'm worried about here: `group1` is defined using an atom from `m`, but we're able to set it as a stereogroup on `m2`. The result is that the following dumps core:
```
        m = Chem.MolFromSmiles('F[C@@H](Br)[C@H](F)Cl |o1:1|')
        m2 = Chem.RWMol(m)

        # Can create a StereoGroup using a Python list
        group1 = Chem.rdchem.StereoGroup(
            Chem.rdchem.StereoGroupType.STEREO_OR, [m.GetAtomWithIdx(1)])
        # Can set the StereoGroups using a Python list
        m2.SetStereoGroups([group1])
        self.assertEqual(len(m2.GetStereoGroups()), 1)

        m = None
        group1 = None
        gc.collect()

        sgs = m2.GetStereoGroups()
        self.assertEqual(len(sgs), 1)
        stereo_atoms = sgs[0].GetAtoms()
        self.assertEqual(stereo_atoms[0].GetIdx(), 1)
        self.assertEqual(stereo_atoms[0].GetOwningMol().GetNumAtoms(), 6)
```

Reply:
It's user error, but it really shouldn't be possible to trigger a core dump from Python

Reply:
yikes! Thanks.